internal/encoding/parquet/parquet.go

```diff
@@ -0,0 +1,125 @@
+package parquet
+
+import (
+	"bytes"
+	goparquet "github.com/fraugster/parquet-go"
+	"github.com/kelindar/talaria/internal/encoding/typeof"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+	"io"
+	"os"
+	"sort"
+)
+
+var errNoWriter = errors.New("unable to create Parquet writer")
+
+// Iterator represents parquet data frame.
+type Iterator interface {
+	io.Closer
+	Range(f func(int, []interface{}) bool, columns ...string) (int, bool)
+	Schema() typeof.Schema
+}
+
+// FromFile creates an iterator from a file.
+func FromFile(filename string) (Iterator, error) {
+	rf, err := os.Open(filename)
+
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := goparquet.NewFileReader(rf)
+	return &iterator{reader: r}, nil
+}
+
+// FromBuffer creates an iterator from a buffer.
+func FromBuffer(b []byte) (Iterator, error) {
+	r, err := goparquet.NewFileReader(bytes.NewReader(b))
+	if err != nil {
+		return nil, err
+	}
+
+	return &iterator{reader: r}, nil
+}
+
+// Range is a helper function that ranges over a set of columns in a Parquet buffer
+func Range(payload []byte, f func(int, []interface{}) bool, columns ...string) error {
+	i, err := FromBuffer(payload)
+	if err != nil {
+		return err
+	}
+
+	_, _ = i.Range(f, columns...)
+	return nil
+}
+
+// First selects a first row only, then stops.
+func First(payload []byte, columns ...string) (result []interface{}, err error) {
+	err = Range(payload, func(_ int, v []interface{}) bool {
+		result = v
+		return true // No need to iterate further, we just take 1st element
+	}, columns...)
+	return
+}
+
+// Iterator represents parquet data frame.
+type iterator struct {
+	reader *goparquet.FileReader
+}
+
+// Range iterates through the reader.
+func (i *iterator) Range(f func(int, []interface{}) bool, columns ...string) (index int, stop bool) {
+	//TODO: Do this once the release is done
```

Comment:
Is this `TODO` still relevant?

Reply:
Yes, waiting for the Parquet library we are using to accept my PR and do a new release